From 667bea7d7c677d1847e6327fccce25ecaa0c8947 Mon Sep 17 00:00:00 2001
From: Jason Wilkes <letshaveanadventure@gmail.com>
Date: Thu, 18 Dec 2014 18:10:15 -0800
Subject: [PATCH] fs: binfmt_misc: fix regression with systemd

Systemd fails to start binfmt.service after the recent
changes to fs/binfmt_misc.c. This reverts the changes
that cause those regressions, while keeping everything else.
---
 fs/binfmt_misc.c | 5 +++--
 1 file changed, 3 insertions(+), 2 deletions(-)

diff --git a/fs/binfmt_misc.c b/fs/binfmt_misc.c
index c04ef1d..4e30fcc 100644
--- a/fs/binfmt_misc.c
+++ b/fs/binfmt_misc.c
@@ -23,6 +23,7 @@
 #include <linux/mount.h>
 #include <linux/syscalls.h>
 #include <linux/fs.h>
+
 #include <linux/uaccess.h>
 
 #ifdef DEBUG
@@ -379,7 +380,7 @@ static Node *create_entry(const char __user *buffer, size_t count)
 		if (!p)
 			goto einval;
 		p[-1] = '\0';
-		if (p == e->magic)
+		if (!e->magic[0])
 			goto einval;
 		if (USE_DEBUG)
 			print_hex_dump_bytes(
@@ -392,7 +393,7 @@ static Node *create_entry(const char __user *buffer, size_t count)
 		if (!p)
 			goto einval;
 		p[-1] = '\0';
-		if (p == e->mask) {
+		if (!e->mask[0]) {
 			e->mask = NULL;
 			pr_debug("register:  mask[raw]: none\n");
 		} else if (USE_DEBUG)
-- 
2.2.0

